### ISSUE TITLE
sidebar: add ability to load a web site into the sidebar

### ISSUE THREAD

**Issue by 424344** [brave/brave-browser#22594]
please add the ability to load a website into the sidebar. 
a custom button in the sidebar should be added to show that page in the sidebar.

## Steps to Reproduce

   1. 
   2. 
   3. 

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by diracdeltas** [comment]
do you mean load the page inside the sidebar, or simply add a button that opens the page in a normal tab? the sidebar currently has no security indicators, so those would need to be added before it's safe to load an arbitrary page inside the sidebar.

**Comment by 424344** [comment]
I mean load the entire page inside the sidebar. 
There are some sites that I'd like to have visible in the sidebar of brave, like https://bookmarkos.com
This is a screenshot of Bookmark OS loaded in side panel in firefox 8I want to do the same in Brave)
![image](https://user-images.githubusercontent.com/26486758/165716115-84458450-f468-4945-a66d-ce624f3f5c75.png)

**Comment by NumDeP** [comment]
@424344 in what other kind of aspect could this beneficial other than that one screenshot example?

**Comment by NumDeP** [comment]
@424344 in what other of aspect could this be beneficial other than that one screenshot example?

**Comment by NumDeP** [comment]
@424344 in what other of aspect could this be beneficial other than that one screenshot example because I presume you'd have to eventually fully load the site on the sidebar?

**Comment by 424344** [comment]
websites must be able to detect screen size, and load an appropriate version (ex the mobile version, not the full site version).
maybe brave could implement an option to automatically request the mobile version (the user can choose to enable it or not).
another website I'd like to load into sidebar is telegram. or other messengers too, or any website that I'd like to see in any moment while browsing in the normal pages.

**Comment by UjCbFwtBayFM** [comment]
I was disappointed Brave did not follow the same side-loading features on Edge or Vivaldi. I support this request.

**Comment by metzger100** [comment]
Tell me if I should also open a new Issue(s) for that, because there are more requests in that post, than the one mentioned here in the Issue: 
I liked this Concept Image I saw on the [Community site](https://community.brave.com/t/improved-ui-visual-concept/61790).
![image](https://user-images.githubusercontent.com/62517154/171515596-1ffc67a8-fff8-41bf-b18c-8cb6221b21f0.png)
Whishes of the User: 
- sidebar for mobile websites (like on Vivaldi)
- split screen to work with several websites (like on Maxthon)
- 1-click pop-out video to save your time (like on Opera)
- click tab to scroll top (like on Opera or Yandex)
- ability to chose or pin images for a new tab (because some of them look awesome)

**Comment by niutech** [comment]
Please allow opening siddebar webapps in an overlay instead of a current tab, like in Opera:

![Opera sidebar](https://user-images.githubusercontent.com/384997/176428362-517639bb-723e-46bf-8083-081e30d21d84.png)

Also please save their state upon opening, like in issue #23032.

**Comment by Axenide** [comment]
Any advances on this? I think this feature would be really useful.

**Comment by niutech** [comment]
Still it is not an overlay, but merely a side-by-side view, thus forcing recalculating layout of the visited web page upon opening and closing a sidebar.

**Comment by lazymonkey2** [comment]
> Still it is not an overlay, but merely a side-by-side view, thus forcing recalculating layout of the visited web page upon opening and closing a sidebar.

this is exactly what I want (I don't want an overlay).

**Comment by niutech** [comment]
@lazymonkey2 For a regular sidebar you can already use a web extension like [Page Sidebar](https://chromewebstore.google.com/detail/page-sidebar-open-any-pag/gkkebamcfeaggmcfciekfakbmlgckdnh). Overlay a la Opera with an option to pin it like a regular sidebar would be much more useful.

**Comment by lazymonkey2** [comment]
@niutech I've already tried some extensions and all have limitations.
I need native implementation.
But I can use an overlay sidebar with an option to pin.

**Comment by rimopa** [comment]
I'm a Vivaldi user, and I think their implementation of this feature is very neat.
Here's a little demonstration I made: https://youtu.be/WzIP8Y9Blwg

Some other details that may be useful to consider when implementing it, to decide whether it's best to follow it or to do it differently:
- I didn't show it, but most extensions also work inside the sidebar.
- "new tab" links are opened in the main view.
- When sidebar left + vertical tabs left, the sidebar goes left, tabs at the right of it.

**Comment by niutech** [comment]
@rimopa Yes, but an option to make a sidebar as an overlay a la Opera would be even better.

**Comment by rimopa** [comment]
> @rimopa Yes, but an option to make a sidebar as an overlay a la Opera would be even better.

Hm, I think that that's an interesting idea. For my use case, I think it's more comfortable to use side-by-side ones, where the page layout is always recalculated.

Either way, the original petition of the author (as stated 4 years ago, and restated 6 months ago) is side-by-side, not overlay/pinnable sidebar pages.

I think the petition of an option to make them overlayed over the main page and pinnable should be made into a separate issue.

**Comment by niutech** [comment]
@rimopa You can already tile pages side-by-side by `brave://flags/#brave-split-view` in #38710. The use case for overlaid sidebar is to quickly peek to a website in a sidebar, without resizing and recalculating layout of the main webview on each open/close of the sidebar.
